src/Serilog.Sinks.Email/Sinks/Email/MailKitEmailSink.cs

```diff
@@ -0,0 +1,169 @@
+            : base(batchSizeLimit, period)
+        {
+            if (connectionInfo == null) throw new ArgumentNullException(nameof(connectionInfo));
+
+            _connectionInfo = connectionInfo;
+            _fromAddress = MimeKit.MailboxAddress.Parse(_connectionInfo.FromEmail);
+            _toAddresses = connectionInfo
+                .ToEmail
+                .Split(",;".ToCharArray(), StringSplitOptions.RemoveEmptyEntries)
+                .Select(MimeKit.MailboxAddress.Parse)
+                .ToArray();
+
+            _textFormatter = textFormatter;
+        }
+        
+        private MimeKit.MimeMessage CreateMailMessage(string payload)
+        {
+            var mailMessage = new MimeKit.MimeMessage();
+            mailMessage.From.Add(_fromAddress);
+            mailMessage.To.AddRange(_toAddresses);
+            mailMessage.Subject = _connectionInfo.EmailSubject;
+            mailMessage.Body = _connectionInfo.IsBodyHtml
+                ? new MimeKit.BodyBuilder { HtmlBody = payload }.ToMessageBody()
+                : new MimeKit.BodyBuilder { TextBody = payload }.ToMessageBody();
+            return mailMessage;            
+        }
+
+        protected override void EmitBatch(IEnumerable<LogEvent> events)
+        {
+            if (events == null)
+                throw new ArgumentNullException(nameof(events));
+
+            var payload = new StringWriter();
+
+            foreach (var logEvent in events)
+            {
+                _textFormatter.Format(logEvent, payload);
+            }
+
+            var mailMessage = CreateMailMessage(payload.ToString());
+
+            try
+            {
+                using (var smtpClient = OpenConnectedSmtpClient())
+                {
+                    smtpClient.Send(mailMessage);
+                    smtpClient.Disconnect(quit: false);
+                }
+            }
+            catch(Exception ex)
+            {
+                SelfLog.WriteLine("Failed to send email: {0}", ex.ToString());
+            }
+        }
+
+        /// <summary>
+        /// Emit a batch of log events, running asynchronously.
+        /// </summary>
+        /// <param name="events">The events to emit.</param>
+        /// <remarks>Override either <see cref="PeriodicBatchingSink.EmitBatch"/> or <see cref="PeriodicBatchingSink.EmitBatchAsync"/>,
+        /// not both.</remarks>
+        protected override async Task EmitBatchAsync(IEnumerable<LogEvent> events)
+        {
+            if (events == null)
+                throw new ArgumentNullException(nameof(events));
+
+            var payload = new StringWriter();
+
+            foreach (var logEvent in events)
+            {
+                _textFormatter.Format(logEvent, payload);
+            }
+
+            var mailMessage = CreateMailMessage(payload.ToString());
+
+            try
+            {
+                using (var smtpClient = OpenConnectedSmtpClient())
```

Comment:
Only one of the two `EmitBatch()` methods needs to be overridden - if the API supports `async` then flowing it throughout via the other overidde below should be the way to go. Is that what you're thinking?